expression/aggregation.go

```diff
@@ -31,17 +31,26 @@ type AggregationFunction interface {
 	// Update during executing.
 	Update(row []types.Datum, groupKey []byte, ctx context.Context) error
 
+	// StreamUpdate updates data using streaming algo.
+	StreamUpdate(row []types.Datum, groupKey []byte, ctx context.Context) (bool, error)
```

Comment:
But currently we didn't apply stream alogrithm to distinct function, so you see there are the same logics in stream update and non-stream update.
If we implement stream distinct function, we will also write two different updates for them :)